refinery/core/models.py

```diff
@@ -593,37 +593,38 @@ def update_investigation(self, investigation, message):
         return version + 1
 
     def get_version(self):
-        try:
-            version = (
-                InvestigationLink.objects.filter(
-                    data_set=self
-                ).aggregate(Max("version"))["version__max"]
-            )
-            return version
-        except:
-            return None
+        latest_investigation_link = self.get_latest_investigation_link()
+        if latest_investigation_link is not None:
+            return latest_investigation_link.version
+        return None
 
     def get_latest_investigation_link(self, version=None):
         """
         Try to fetch the latest InvestigationLink related to the DataSet
         instance. If a version is provided, try to fetch the latest
         InvestigationLink for said version.
         :param version: integer
-        :returns: an InvestigationLink Instance or None if Exception occurs
+        :returns: an InvestigationLink Instance or `None`
         """
-        try:
-            if version is None:
-                version = InvestigationLink.objects.filter(
-                        data_set=self
-                    ).aggregate(Max("version"))["version__max"]
-
-            return InvestigationLink.objects.get(
-                data_set=self,
-                version=version
-            )
-        except:
+        if not self.is_valid:
             return None
 
+        if version is None:
+            try:
+                return InvestigationLink.objects.filter(
+                    data_set=self
+                ).latest('date')
+            except InvestigationLink.DoesNotExist as exc:
+                logger.error("Couldn't properly fetch "
+                             "latest InvestigationLink: %s", exc)
```

Comment:
"Failed to retrieve" instead of "Couldn't properly fetch"
https://github.com/refinery-platform/refinery-platform/wiki/Coding-Style-Guide#meta